lib/android-helpers.js

```diff
@@ -476,22 +476,36 @@ helpers.setMockLocationApp = async function setMockLocationApp (adb, app) {
   }
 };
 
-helpers.installHelperApp = async function installHelperApp (adb, apkPath, packageId, appName) {
-  try {
+helpers.installHelperApp = async function installHelperApp (adb, apkPath, packageId) {
+  // Sometimes adb push or adb instal take more time than expected to install an app
+  // e.g. https://github.com/appium/io.appium.settings/issues/40#issuecomment-476593174
+  await retry(2, async function retryInstallHelperApp () {
     await adb.installOrUpgrade(apkPath, packageId, {grantPermissions: true});
-  } catch (err) {
-    logger.warn(`Ignored error while installing Appium ${appName} helper: ` +
-                `'${err.message}'. Manually uninstalling the application ` +
-                `with package id '${packageId}' may help. Expect some Appium ` +
-                `features may not work as expected unless this problem is ` +
-                `fixed.`);
-  }
+  });
 };
 
+/**
+ * Pushes and installs io.appium.settings app.
+ * Throws an error if the setting app is required
+ *
+ * @param {Adb} adb - The adb module instance.
+ * @param {boolean} throwError[false] - Whether throw error or not
+ * @throws {Error} If throwError is true and something happens in installation step
+ */
 helpers.pushSettingsApp = async function pushSettingsApp (adb, throwError = false) {
   logger.debug('Pushing settings apk to device...');
 
-  await helpers.installHelperApp(adb, settingsApkPath, SETTINGS_HELPER_PKG_ID, 'Settings');
+  try {
+    await helpers.installHelperApp(adb, settingsApkPath, SETTINGS_HELPER_PKG_ID, throwError);
+  } catch (err) {
+    if (throwError) {
+      throw err;
+    }
+
+    logger.warn(`Ignored error while installing ${settingsApkPath} fails: ` +
+                `'${err.message}'. Features require the apk such as toggle WiFi ` +
```

Comment:
Features that rely on this helper ... will raise an error if you try to use them